components/core/src/clp/streaming_compression/lzma/Decompressor.hpp

```diff
@@ -0,0 +1,105 @@
+#ifndef CLP_STREAMING_COMPRESSION_LZMA_DECOMPRESSOR_HPP
+#define CLP_STREAMING_COMPRESSION_LZMA_DECOMPRESSOR_HPP
+
+#include <cstddef>
+
+#include "../../ErrorCode.hpp"
+#include "../../ReaderInterface.hpp"
+#include "../../TraceableException.hpp"
+#include "../Constants.hpp"
+#include "../Decompressor.hpp"
+
+namespace clp::streaming_compression::lzma {
+class Decompressor : public ::clp::streaming_compression::Decompressor {
+public:
+    // Types
+    class OperationFailed : public TraceableException {
+    public:
+        // Constructors
+        OperationFailed(ErrorCode error_code, char const* const filename, int line_number)
+                : TraceableException(error_code, filename, line_number) {}
+
+        // Methods
+        [[nodiscard]] auto what() const noexcept -> char const* override {
+            return "streaming_compression::lzma::Decompressor operation failed";
+        }
+    };
+
+    // Constructor
+    Decompressor() : clp::streaming_compression::Decompressor(CompressorType::LZMA) {}
+
+    // Destructor
+    ~Decompressor() override = default;
+
+    // Delete copy constructor and assignment operator
+    Decompressor(Decompressor const&) = delete;
+    auto operator=(Decompressor const&) -> Decompressor& = delete;
+
+    // Delete move constructor and assignment operator
+    // TODO: change to default when the base decompressor class has been updated
+    Decompressor(Decompressor&&) noexcept = delete;
+    auto operator=(Decompressor&&) noexcept -> Decompressor& = delete;
+
+    // Methods implementing the ReaderInterface
+    /**
+     * Tries to read up to a given number of bytes from the decompressor.
+     * @param buf
+     * @param num_bytes_to_read The number of bytes to try reading
+     * @param num_bytes_read The actual number of bytes read
```

Comment:
```suggestion
     * @param buf
     * @param num_bytes_to_read The number of bytes to try reading.
     * @param num_bytes_read The actual number of bytes read.
```
Let's be consistent to add `.` after:
- Description
- param/return
- Throw

Do it for the rest